cmd/logs.go

```diff
@@ -0,0 +1,307 @@
+
+	return cmd
+}
+
+func logs(cmd *cobra.Command, token string, workflow string) error {
+	filters, _ := cmd.Flags().GetStringSlice("filter")
+	jsonOutput, _ := cmd.Flags().GetBool("json")
+	page, _ := cmd.Flags().GetInt64("page")
+	size, _ := cmd.Flags().GetInt64("size")
+
+	steps, chosenFilters, err := parseLogsFilters(filters)
+	if err != nil {
+		return err
+	}
+
+	logsParams := operations.NewGetWorkflowLogsParams()
+	logsParams.SetAccessToken(&token)
+	logsParams.SetWorkflowIDOrName(workflow)
+	logsParams.SetPage(&page)
+	logsParams.SetSteps(steps)
+	if cmd.Flags().Changed("size") {
+		logsParams.SetSize(&size)
+	}
+
+	api, err := client.ApiClient()
+	if err != nil {
+		return err
+	}
+	logsResp, err := api.Operations.GetWorkflowLogs(logsParams)
+	if err != nil {
+		return fmt.Errorf("workflow logs could not be retrieved:\n%v", err)
+	}
+
+	var workflowLogs Logs
+	err = json.Unmarshal([]byte(logsResp.GetPayload().Logs), &workflowLogs)
+	if err != nil {
+		return err
+	}
+
+	err = filterJobLogs(&workflowLogs.JobLogs, chosenFilters)
+	if err != nil {
+		return err
+	}
+
+	if jsonOutput {
+		err := utils.DisplayJsonOutput(workflowLogs, cmd.OutOrStdout())
+		if err != nil {
+			return err
+		}
+	} else {
+		displayHumanFriendlyLogs(cmd, workflowLogs, steps)
+	}
+
+	return nil
+}
+
+func parseLogsFilters(filters []string) ([]string, map[string]string, error) {
+	availableFilters := map[string]string{
+		"step":            "job_name",
+		"compute_backend": "compute_backend",
+		"docker_img":      "docker_img",
+		"status":          "status",
+	}
+
+	var steps []string
+	chosenFilters := make(map[string]string)
+	for _, filter := range filters {
+		filterName, filterValue, err := utils.GetFilterNameAndValue(filter)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		_, isFilterValid := availableFilters[filterName]
+		if !isFilterValid {
+			validFilters := make([]string, 0, len(availableFilters))
+			for key := range availableFilters {
+				validFilters = append(validFilters, key)
+			}
+			sort.Strings(validFilters)
+
+			return nil, nil, fmt.Errorf(
+				"filter '%s' is not valid\nAvailable filters are '%s'",
+				filterName,
+				strings.Join(validFilters, "' '"),
+			)
+		}
+
+		if filterName == "step" {
+			steps = append(steps, filterValue)
+		} else {
+			realValue, isKnownBackend := utils.ReanaComputeBackends[strings.ToLower(filterValue)]
+			if filterName == "compute_backend" && isKnownBackend {
+				filterValue = realValue
+			} else if filterName == "status" && !slices.Contains(utils.GetRunStatuses(true), filterValue) {
+				return nil, nil, fmt.Errorf("input status value '%s' is not valid", filterValue)
+			}
+			realFilterName := availableFilters[filterName]
+			chosenFilters[realFilterName] = filterValue
+		}
+	}
+
+	return steps, chosenFilters, nil
+}
+
+func displayHumanFriendlyLogs(cmd *cobra.Command, logs Logs, steps []string) {
+	leadingMark := "==>"
+
+	if logs.WorkflowLogs != nil {
```

Comment:
It seems like that sometimes `logs.WorkflowLogs` can be the empty string, for example when checking the logs of a workflow that is in the created status. In this case, the Python client does not show anything, while the Go client only shows `==> Workflow engine logs`. Can you reproduce this issue?

Reply:
Yes I can reproduce it. Can this also happen for EngineSpecific? I can check for the empty string just in case